packages/lodestar/src/api/impl/validator/index.ts

```diff
@@ -58,6 +59,7 @@ const SYNC_TOLERANCE_EPOCHS = 1;
  */
 export function getValidatorApi({chain, config, logger, metrics, network, sync}: ApiModules): routes.validator.Api {
   let genesisBlockRoot: Root | null = null;
+  const nextEpochProposerDutyCache = new Map<Epoch, ValidatorIndex[]>();
```

Comment:
Keeping caches here is not a good separation of concerns. We must keep all caches attached to the BeaconChain class, this functions should be state-less